src/Microsoft.Health.Dicom.SqlServer/Features/Store/SqlIndexDataStore.cs

```diff
@@ -48,6 +56,16 @@ public async Task<long> CreateInstanceIndexAsync(DicomDataset instance, Cancella
             using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionFactoryWrapper.ObtainSqlConnectionWrapperAsync(cancellationToken))
             using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
             {
+                IEnumerable<InsertStringCustomTagTableTypeV1Row> stringCustomTags =
+                    DetermineKeysAndTagsForDataType(storedCustomTagEntries, CustomTagDataType.StringData).Select(x => new InsertStringCustomTagTableTypeV1Row(x.Key, instance.GetString(x.Value)));
+                IEnumerable<InsertBigIntCustomTagTableTypeV1Row> bigIntCustomTags =
+                    DetermineKeysAndTagsForDataType(storedCustomTagEntries, CustomTagDataType.LongData).Select(x => new InsertBigIntCustomTagTableTypeV1Row(x.Key, long.Parse(instance.GetString(x.Value))));
+                IEnumerable<InsertDoubleCustomTagTableTypeV1Row> doubleCustomTags =
+                    DetermineKeysAndTagsForDataType(storedCustomTagEntries, CustomTagDataType.DoubleData).Select(x => new InsertDoubleCustomTagTableTypeV1Row(x.Key, long.Parse(instance.GetString(x.Value))));
+                IEnumerable<InsertDateTimeCustomTagTableTypeV1Row> dateTimeCustomTags =
+                    DetermineKeysAndTagsForDataType(storedCustomTagEntries, CustomTagDataType.DateTimeData).Select(x => new InsertDateTimeCustomTagTableTypeV1Row(x.Key, instance.GetStringDateAsDateTime(x.Value).Value));
+                IEnumerable<InsertPersonNameCustomTagTableTypeV1Row> personNameCustomTags =
```

Comment:
should we update https://github.com/microsoft/dicom-server/blob/main/src/Microsoft.Health.Dicom.Core/Features/Store/DicomDatasetValidator.cs to validate custom tags?

Reply:
+1